tests/tests_tangd_custom_port.yml

```diff
@@ -0,0 +1,48 @@
+---
+- name: Test tangd_custom_port
+  hosts: all
+  vars:
+    nbde_server_port: 7500
+    nbde_server_firewall_zone: public
+  tasks:
+    - name: install with custom port and firewall zone
+      import_role:
+        name: linux-system-roles.nbde_server
+
+    - name: check if port is open
+      shell:
+        cmd: |-
+          set -euo pipefail
+          ss -tulpn | grep {{ nbde_server_port }} | awk -F' ' '{print $5}'
+      register: __open_ports_output
+      failed_when: __open_ports_output.stdout != "*:{{ nbde_server_port }}"
+
+    - name: check if port TCP is open
+      shell:
+        cmd: |-
+          set -euo pipefail
+          ss -tulpn | grep {{ nbde_server_port }} | awk -F' ' '{print $1}'
+      register: __open_ports_output
+      failed_when: __open_ports_output.stdout != "tcp"
+
+    - name: check if port is opened in firewall
+      shell:
+        cmd: |-
+          set -euo pipefail
+          firewall-cmd --list-all | grep {{ nbde_server_port }}/tcp | \
+            awk -F':' '{print $2}' | \
+            awk '{gsub(/^[ \t]+| [ \t]+$/,""); print$0}'
+      register: __firewall_output
+      failed_when: __firewall_output.stdout != "{{ nbde_server_port }}/tcp"
+
+    - name: check if firewall zone is set
```

Comment:
I don't think you need this check at all - the `firewall-cmd --zone {{ nbde_server_firewall_zone }} --query-port {{ nbde_server_port }}/tcp` will check that the given zone exists and that the port is set in the given zone.